docs/proposals/non-default-namespace-deployment.md

```diff
@@ -0,0 +1,145 @@
+# Proposal: Supporting non default namespace CSI deployment at Openshift 
+
+Last updated: 09.12.2021
+
+## Abstract
+
+This proposal contains approaches for supporting CSI deployment in non default namespace at Openshift platform, 
+introducing Privileged SCC at CSI ServiceAccounts.`
+
+## Background
+
+In case of CSI deployment at Openshift platform in non default namespace, CSI pods have Restricted SCC by default.   
+But restricted denies access to all host features and resources like disk. Must be Privileged SCC because it allows
+access to all privileged and host features and resources.
+For give privileged SCC to custom user, the following command need to be executed:
+`oc adm policy add-scc-to-user privileged -z <user name> -n <namespace>`
+And so, we should give this scc for CSI's service accounts, that need it: 
+ * csi-node-sa;
+ * csi-baremetal-extender-sa.
```

Comment:
Why do we need to have privileged access for `extender-sa` on OpenShift? There is no `patcher` component deployed there

Reply:
Because scheduler extender is deployed at openshift:
https://github.com/dell/csi-baremetal-operator/blob/master/pkg/scheduler_extender.go#L100

Reply:
Not sure what resources managed by `scheduler extender` require privileged access

Reply:
It needs:
1. Hostnetwork;
2. Hostports.

Reply:
Ok, thanks! Looks like we need to get rid of this and use kubernetes service instead